src/module.c

```diff
@@ -1062,7 +1085,11 @@ int64_t commandFlagsFromString(char *s) {
         else if (!strcasecmp(t,"no-cluster")) flags |= CMD_MODULE_NO_CLUSTER;
         else if (!strcasecmp(t,"no-mandatory-keys")) flags |= CMD_NO_MANDATORY_KEYS;
         else if (!strcasecmp(t,"allow-busy")) flags |= CMD_ALLOW_BUSY;
-        else break;
+        else if (!matchAclCategoriesFlags(t, acl_categories));
+        else {
+            serverLog(LL_WARNING,"Unrecognized command flag %s on module load", t);
+            break;
+        }
```

Comment:
`matchAclCategoriesFlags` returns REDISMODULE_OK/REDISMODULE_ERR so `!` should not be used here (the reason it worked is that OK is 0 and ERR isn't)
regardless, i find this code structure a bit confusing, why not:
```
else {
    if (matchAclCategoriesFlags(t, acl_categories) == REDISMODULE_ERR) {
        serverLog(LL_WARNING,"Unrecognized command flag %s on module load", t);
        break;
   }
}
```

Reply:
I'm fine with the new structure, but I don't really think matchAclCategoriesFlags should return REDISMODULE_ERR or REDISMODULE_OK now that I think about it, since those are more typically used to module error codes. Since this is an internal function, it should probably directly return 1 and 0.